modules/firefox/hm.nix

```diff
@@ -1,36 +1,66 @@
-# Consider also updating the LibreWolf module when updating this module,
-# as they are very similar.
-
-{ config, lib, ... }:
+{ config, lib, ...}:
 
 let
-  profileSettings = {
-    settings = {
-      "font.name.monospace.x-western" = config.stylix.fonts.monospace.name;
-      "font.name.sans-serif.x-western" = config.stylix.fonts.sansSerif.name;
-      "font.name.serif.x-western" = config.stylix.fonts.serif.name;
-    };
-  };
-  makeProfileSettingsPair = profileName:
-    lib.nameValuePair profileName profileSettings;
-  derivatives = [
-    {path = "firefox"; name = "Firefox";}
-    {path = "librewolf"; name = "LibreWolf";}
+  targets = [
+    { path = "firefox"; name = "Firefox"; }
+    { path = "librewolf"; name = "LibreWolf"; }
   ];
+  eachConfig = mkCfg: targets: lib.mkMerge (map mkCfg targets);
+  eachTarget = mkCfg: lib.mkIf config.stylix.enable (eachConfig (target: let
+    cfg = config.stylix.targets.${target.path};
+    programCfg = config.programs.${target.path};
+  in
+    lib.mkIf cfg.enable (mkCfg {inherit target cfg programCfg;})
+  ) targets);
 in {
-  options.stylix.targets = lib.listToAttrs (map (drv: lib.nameValuePair drv.path {
-    enable =
-      config.lib.stylix.mkEnableTarget drv.name true;
+  options.stylix.targets = lib.listToAttrs (map (target:
+    lib.nameValuePair target.path {
+      enable = config.lib.stylix.mkEnableTarget target.name true;
+
+      profileNames = lib.mkOption {
+        description = "The ${target.name} profile names to apply styling on.";
+        type = lib.types.listOf lib.types.str;
+        default = [];
+      };
+
+      firefoxGnomeTheme.enable = config.lib.stylix.mkEnableTarget "Firefox GNOME theme" false;
```

Comment:
Should I enable the theme by default for better discoverability?

Reply:
No. The GNOME theme is merely an alternative to the default Firefox theming. Plus:

> > [!WARNING]
> > ### Disclaimer:
> > Be aware that this theme might do things that are not supported by upstream Firefox. If you face an issue while using this theme, report it here first or test if it is reproducible in vanilla Firefox.
> >
> > If you are a software distribution maintainer, please do not ship these changes by default to your users unless you make it extremely clear that they are using a modified version of the Firefox UI.
>
> [firefox-gnome-theme, README.md](https://github.com/rafaelmardojai/firefox-gnome-theme/blob/92f4890bd150fc9d97b61b3583680c0524a8cafe/README.md?plain=1#L15-L19)